Add a new `fromOptionError` operator (#3813)
zio/zio: docs/overview/creating_effects.md

An `Option` can be converted into a ZIO effect using `ZIO.fromOption`:

```scala mdoc:silent
val zoption: IO[Option[Nothing], Int] = ZIO.fromOption(Some(2))
```

> The error type of the resulting effect is `Option[Nothing]`, because the `None` case of `Option` provides no information on why the value is not there. You can change the `Option[Nothing]` into a more specific error type using `ZIO#mapError`:

Reviewer: ```suggestion
The error type of the resulting effect is `Option[Nothing]`, which provides no information on why the value is not there. You can change the `Option[Nothing]` into a more specific error type using `ZIO#mapError`:
```

```scala mdoc:silent
val zoption2: IO[String, Int] = zoption.mapError(_ => "It wasn't there!")
```

You can also readily compose it with other operators while preserving the optional nature of the result (similar to an `OptionT`)

```scala mdoc:silent

> val maybeId: IO[Option[Nothing], String] = ZIO.fromOption(Some("abc123"))

Reviewer: Really nice example!

An `Option` can be converted into a ZIO effect using `ZIO.fromOption`:

```scala mdoc:silent
val zoption: IO[Option[Nothing], Int] = ZIO.fromOption(Some(2))
```

> The error type of the resulting effect is `Unit`, because the `None` case of `Option` provides no information on why the value is not there. You can change the `Unit` into a more specific error type using `ZIO#mapError`:

Reviewer 2: Does this paragraph need to be updated in lieu of the error channel now being defined as `Option[Nothing]` instead of `Unit`?
